build.js

```diff
@@ -1839,8 +1854,14 @@ var dirtyDetectingBuilder = function(rootPackageName, resultsCache, prevResultsC
         autogenAliases.genSourceFiles.internalInterface,
         autogenAliases.genSourceFiles.internalImplementation,
       ])
+      .concat(commonML.linkFlags)
+      // Used only for C dependencies currently. Should be only _already_ built
+      // files.
+      .concat(commonML.foreignDependencies && commonML.foreignDependencies.length > 0
+        ? ['-custom'].concat(commonML.foreignDependencies.map(function(v){ return path.join(packageResource.realPath, v); }))
+        : [])
+      .concat(['-o', autogenAliases.genSourceFiles.internalImplementation.replace('.ml', '.cma')])
```

Comment:
For native targets, this would need to be `.cmxa`. You should probably abstract that away into some helper utility that matches the way it's done in the rest of CommonML.